dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/resourcetable/DefaultResourceTableService.java

```diff
@@ -255,4 +259,31 @@ public void dropAllSqlViews( JobProgress progress )
         progress.startingStage( "Drop SQL views", nonQueryViews.size(), SKIP_ITEM );
         progress.runStage( nonQueryViews, SqlView::getViewName, sqlViewService::dropViewTable );
     }
+
+    @Override
+    public List<Integer> generateDataYears()
+    {
+        List<Integer> availableDataYears = new ArrayList<>();
+
+        availableDataYears.addAll( resourceTableStore.getAvailableDataYears() );
+
+        if ( availableDataYears.isEmpty() )
+        {
+            availableDataYears = new ArrayList<>();
```

Comment:
Do we need to instantiate a new list here?

Reply:
correct, we don't need that.